### PR TITLE
citgm-all: get modules from github by default

### PR DESCRIPTION
**Pull request by gdams**
Currently we pass `"replace": true` into our lookup.json file to get the tests from github instead of npm. At the moment all of our modules is lookup.json contain this tag so it makes no sense to make it an option. 

This PR changes makes replace redundant and adds `"npm": true` to NOT get the tests from github and instead get them from npm. This means that existing users with custom json files will still be able to get from npm or github

### CODE REVIEW

**Comment by TheAlphaNerd** [comment]
I'm not on board with this change as it is. We should not be changing the semantics of replace.

It would make more sense to deprecate the use of replace and introduce a new tag `npm: true`

File: lib/lookup.js [hunk 1]
```diff
@@ -76,20 +76,25 @@ function resolve(context, next) {
       if (rep.stripAnsi) {
         context.module.stripAnsi = rep.stripAnsi;
       }
-      if (rep.replace) {
-        if (!rep.repo && !meta.repository) {
-          next(new Error('no-repository-field in package.json'));
-          return;
+      if (!rep.npm) {
+        if (typeof rep.replace === 'boolean' && rep.replace === false) {
```
Comment (TheAlphaNerd) [inline lib/lookup.js]: `rep.replace === false` can be replaced with `!rep.replace`

File: lib/lookup.js [hunk 2]
```diff
@@ -76,20 +76,25 @@ function resolve(context, next) {
...
+          rep.npm=true;
```
Comment (TheAlphaNerd) [inline lib/lookup.js]: nit: should be spaces on either side of the equality sign here, surprised that didn't fail the linter

File: lib/lookup.js (hunk 1 above)
Comment (TheAlphaNerd) [inline lib/lookup.js]: this should be hoisted outside of this if statement to avoid having to do the equality check twice in a row for `!rep.npm

File: lib/lookup.js [hunk 3]
```diff
@@ -76,20 +76,25 @@ function resolve(context, next) {
...
         }
-        context.emit('info','lookup-found',detail.name);
-        var url = makeUrl(
+        if (!rep.npm){
```
Comment (TheAlphaNerd) [inline lib/lookup.js]: why the second equality check here, see above comment regarding hoisting the replace check to avoid having to check twice

File: lib/lookup.js [hunk 4]
```diff
@@ -76,20 +76,25 @@ function resolve(context, next) {
...
+          if (!rep.repo && !meta.repository) {
+            next(new Error('no-repository-field in package.json'));
+            return;
+          }
+          context.emit('info','lookup-found',detail.name);
+          var url = makeUrl(
           getRepo(rep.repo, meta),
```
Comment (TheAlphaNerd) [inline lib/lookup.js]: nit: indentation here is wrong

File: test/fixtures/custom-lookup-no-repo.json [hunk 5]
```diff
@@ -1,5 +1,4 @@
 {
   "omg-i-pass": {
```
Comment (TheAlphaNerd) [inline test/fixtures/custom-lookup-no-repo.json]: for this case should we perhaps support a module simply being a string?

File: test/fixtures/custom-lookup-script.json [hunk 6]
```diff
@@ -1,9 +1,10 @@
 {
   "omg-i-pass": {
-    "replace": false,
+    "npm": true,
     "script": "./example-test-script-passing.sh"
   },
   "omg-i-have-script": {
+    "npm": true,
```
Comment (TheAlphaNerd) [inline test/fixtures/custom-lookup-script.json]: why did you add `npm: true` here?

File: test/fixtures/pretty-print.json [hunk 7]
```diff
@@ -1,14 +1,14 @@
 {
   "chalk": {
-    "replace": true,
+    "npm": true,
```
Comment (TheAlphaNerd) [inline test/fixtures/pretty-print.json]: why did you substitute `replace: true` with `npm: true`?

File: test/fixtures/pretty-print.json [hunk 8]
```diff
@@ -1,14 +1,14 @@
...
     "prefix": "v"
   },
   "colors": {
-    "replace": true,
```
Comment (TheAlphaNerd) [inline test/fixtures/pretty-print.json]: same as above

File: test/fixtures/pretty-print.json [hunk 9]
```diff
@@ -1,14 +1,14 @@
...
+    "npm": true,
     "prefix": "v"
   },
   "yosay": {
-    "replace": true,
+    "npm": true,
```
Comment (TheAlphaNerd) [inline test/fixtures/pretty-print.json]: same as above

File: test/test-lookup.js [hunk 10]
```diff
@@ -49,7 +49,7 @@ test('lookup[getLookupTable]:', function (t) {
   }
   t.ok(table, 'table should exist');
   t.ok(table.lodash, 'lodash should be in the table');
-  t.ok(table.lodash.replace, 'lodash should need to be replaced');
+  t.ok(table.lodash.skip, 'lodash should contain a skip paramter');
```
Comment (TheAlphaNerd) [inline test/test-lookup.js]: where does the skip parameter come from? I don't believe we have or support a skip parameter do we?

Comment (GeorgeAdams95) [inline test/test-lookup.js]: I just changed it from replace to skip as replace was removed
https://github.com/nodejs/citgm/blob/master/lib/lookup.json#L4

File: test/fixtures/pretty-print.json (hunk 7 above)
Comment (GeorgeAdams95) [inline test/fixtures/pretty-print.json]: `replace:true` does nothing now so I figured that I would just update the code to use `npm:true`

**Comment by TheAlphaNerd** [comment]
this is failing on v0.12

**Comment by gibfahn** [comment]
@TheAlphaNerd I guess we could just wait to merge this until the new year?

**Comment by GeorgeAdams95** [comment]
@TheAlphaNerd @gibfahn It wasn't failing on 0.12 until my last run, i'll look into why

**Comment by GeorgeAdams95** [comment]
@TheAlphaNerd I have rebased on master and the new CI run seemed to pass on node 12.  PTAL

File: lib/lookup.js [hunk 11]
```diff
@@ -76,18 +76,21 @@ function resolve(context, next) {
...
+        rep.npm = true;
+      }
+      if (!rep.npm){
         if (!rep.repo && !meta.repository) {
           next(new Error('no-repository-field in package.json'));
           return;
         }
         context.emit('info','lookup-found',detail.name);
         var url = makeUrl(
-          getRepo(rep.repo, meta),
```
Comment (TheAlphaNerd) [inline lib/lookup.js]: why did you remove the indentation from here?

File: test/fixtures/custom-lookup-script.json (hunk 6 above)
Comment (TheAlphaNerd) [inline test/fixtures/custom-lookup-script.json]: ???

File: test/fixtures/pretty-print.json (hunk 7 above)
Comment (TheAlphaNerd) [inline test/fixtures/pretty-print.json]: please keep the tests in their original state. Only change if it is for an explicit reason

File: test/test-lookup.js [hunk 12]
```diff
@@ -49,7 +49,7 @@ test('lookup[getLookupTable]:', function (t) {
   }
   t.ok(table, 'table should exist');
   t.ok(table.lodash, 'lodash should be in the table');
-  t.ok(table.lodash.replace, 'lodash should need to be replaced');
+  t.ok(table.underscore.master, 'underscore should contain a master paramter');
```
Comment (TheAlphaNerd) [inline test/test-lookup.js]: why was this switched the underscore?

File: test/fixtures/custom-lookup-script.json (hunk 6 above)
Comment (GeorgeAdams95) [inline test/fixtures/custom-lookup-script.json]: because if it was false then it should be getting it from npm?

File: test/test-lookup.js (hunk 12 above)
Comment (TheAlphaNerd) [inline test/test-lookup.js]: we still are supposed to be supporting `replace` just undocumented, are we not?

**Comment by GeorgeAdams95** [comment]
@TheAlphaNerd nits fixed, PTAL

**Comment by TheAlphaNerd** [comment]
afaict there is still not test for replace. Could you also answer my questions regarding the switch to underscore.

File: test/bin/test-citgm-all.js [hunk 13]
```diff
@@ -72,6 +72,17 @@ test('citgm-all: flaky-fail', function (t) {
   });
 });
 
+test('citgm-all: test with replace', function (t) {
+  t.plan(1);
```
Comment (GeorgeAdams95) [inline test/bin/test-citgm-all.js]: @TheAlphaNerd this is the test for replace

File: test/test-lookup.js (hunk 12 above)
Comment (GeorgeAdams95) [inline test/test-lookup.js]: we cannot go adding replace to our lookup.json, The idea is that we are moving away from using replace. Also all this test does is checks if the tag exists, NOT whether it works

**Comment by GeorgeAdams95** [comment]
@TheAlphaNerd updated PTAL

**Comment by GeorgeAdams95** [comment]
@TheAlphaNerd I have rebased this now on master

**Comment by TheAlphaNerd** [comment]
it just dawned on me that this is Semver Major no matter what we do as it changes the behavior of `citgm module` when it isn't in the lookup.

**Comment by GeorgeAdams95** [comment]
@TheAlphaNerd I was under the impression that this was not the case, i'll double check

**Comment by TheAlphaNerd** [comment]
@GeorgeAdams95 nvm... default behavior will still be npm.

Can you rebase against master and we can get this landed

**Comment by GeorgeAdams95** [comment]
@MylesBorins rebased on master, will land if you are happy to go?

**Comment by gibfahn** [comment]
@MylesBorins Changing to `semver-minor`
